### DISCUSSION TITLE
Error: no overload matches 'Lucky::ProtectFromForgery.get_token' with type (HTTP::Server::Context | Nil)

### DISCUSSION THREAD

**Discussion by drhuffman12** [luckyframework/lucky#1659] [Q&A]
I'm trying to update a dockerized example app using Lucky. (See: https://github.com/drhuffman12/example_lucky_app/pull/3.)

When running `docker-compose up`, I'm getting build/test failures with the following error:

```
Run SPEC_VERBOSE=1 crystal spec
Showing last frame. Use --error-trace for full trace.

In lib/lucky/src/lucky/tags/forgery_protection_helpers.cr:23:35

 23 | content: ProtectFromForgery.get_token(context)
                                  ^--------
Error: no overload matches 'Lucky::ProtectFromForgery.get_token' with type (HTTP::Server::Context | Nil)

Overloads are:
 - Lucky::ProtectFromForgery.get_token(context : HTTP::Server::Context)
Couldn't find overloads for these types:
 - Lucky::ProtectFromForgery.get_token(context : Nil)
Error: Process completed with exit code 1.
```

The `docker-compose build` parts passed passed, but had some red text also. I'm not sure, but the red text parts might be affecting the `docker-compose up` parts.

See also CI failures noted in: https://github.com/drhuffman12/example_lucky_app/runs/4995359592?check_suite_focus=true

**Comment by jwoertink** [comment]
It sounds like this is an app that was upgraded, right? I think it's this issue https://github.com/luckyframework/lucky/issues/1526 

Looking at that app, I see it's using Lucky 0.27. I forget if that version made this change or not. If possible, I'd suggest upgrading to Lucky 0.29. You can use LuckyDiff https://luckydiff.com/?from=0.27.2&to=0.29.0 and check through the UPGRADE NOTES https://github.com/luckyframework/lucky/blob/master/UPGRADE_NOTES.md 

If you're unable to upgrade, we can come up with a temp monkey patch solution. Let me know if that works for you.

**Reply by drhuffman12** [reply]
Thanks; I'll check on that, maybe restart it from scratch w/ the newer version of Lucky or use the 27-to-29 notes.
